lib/heuristics/concerns/scheduling_data_initialisation.rb

```diff
@@ -248,8 +248,8 @@ def best_common_tw(set)
         group_tw.delete_if{ |tw1|
           data[:tws_sets].first.none?{ |tw2|
             (tw1[:day_index].nil? || tw2[:day_index].nil? || tw1[:day_index] == tw2[:day_index]) &&
-              (tw1[:start].nil? || tw2[:end].nil? || tw1[:start] <= tw2[:end]) &&
-              (tw1[:end].nil? || tw2[:start].nil? || tw1[:end] >= tw2[:start])
+              (tw2[:end].nil? || tw1[:start] <= tw2[:end]) &&
```

Comment:
```suggestion
              (tw2.end.nil? || tw1.start <= tw2.end) &&
```

Reply:
these timewindows are not from model timewindow (this should be improved in another PR : https://gitlab.com/mapotempo/optimizer-api/-/issues/689)